dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java

```diff
@@ -618,25 +625,44 @@ public boolean isAccountExpired(User user) {
 
   @Override
   @Transactional(readOnly = true)
-  public List<ErrorReport> validateUserCreateOrUpdate(User user, User currentUser) {
+  public List<ErrorReport> validateUserCreateOrUpdateAccess(User user, User currentUser) {
 
     List<ErrorReport> errors = new ArrayList<>();
 
-    if (currentUser == null || user == null) {
+    if (currentUser == null || user == null || currentUser.isSuper()) {
       return errors;
     }
 
     User userToChange = userStore.get(user.getId());
-    if (!currentUser.isSuper() && userToChange != null && userToChange.isSuper()) {
+    if (userToChange != null && userToChange.isSuper()) {
       errors.add(new ErrorReport(User.class, ErrorCode.E3041, currentUser.getUsername()));
     }
 
     validateUserRoles(user, currentUser, errors);
     validateUserGroups(user, currentUser, errors);
 
+    validateOrgUnit(user.getOrganisationUnits(), currentUser, errors);
+    validateOrgUnit(user.getDataViewOrganisationUnits(), currentUser, errors);
+    validateOrgUnit(user.getTeiSearchOrganisationUnits(), currentUser, errors);
+
     return errors;
   }
 
+  private void validateOrgUnit(
```

Comment:
I think either the method needs a more intention carrying name or a comment to explain what we try to prevent with this check. From the name alone as it is this is very hard to reverse engineer.